server/moodle/mod/livequiz/amd/src/delete_question.js

```diff
@@ -16,18 +16,23 @@ export function add_delete_question_listeners(quizid, lecturerid) {
     element.addEventListener("click", () => {
       let list_item = element.closest("li"); // Get the list item.
       let questionid = parseInt(element.dataset.id, 10);
-            if(!confirm("Are you sure you want to delete this question?")){
+            if (!confirm("Are you sure you want to delete this question?")) {
                 return;
-            } else{
-                delete_question(questionid, lecturerid, quizid);
-                list_item.remove(); // Remove the question from the list.
-                element.remove(); // Remove the delete button.
+            } else {
+                delete_question(questionid, lecturerid, quizid)
+                    .then((response) => {
+                        if (response.success) {
+                            list_item.remove(); // Remove the question from the list.
+                            element.remove(); // Remove the delete button.
+                            let updated_list_length = question_list.querySelectorAll("li").length;
+                            if (updated_list_length === 0) {
+                                rerender_take_quiz_button(take_quiz_url, false);
+                            }
+                        } else {
+                            alert("Cannot delete question: " + response.message);
```

Comment:
The response for this should be more saying. Right now its "Cannot delete **answer** with participations". @Peterrahr is on it on a new PR